rules/jsll/src/rules/jsll-awa-init/jsll-awa-init.ts

```diff
@@ -23,89 +23,136 @@ const debug: debug.IDebugger = d(__filename);
 const rule: IRuleBuilder = {
     create(context: RuleContext): IRule {
         const linter = new Linter();
-        let validated: boolean = false; // If `validateAwaInit` has run.
-        let hasJSLLScript: boolean = false; // If link to JSLL scripts has been included.
+        /** States available. */
+        const states = {
+            apiBody: 'apiBody',
+            apiHead: 'apiHead',
+            body: 'body',
+            bodyOtherScript: 'bodyOtherScript',
+            head: 'head',
+            headOtherScript: 'headOtherScript',
+            start: 'start'
+        };
+        /** Current state. */
+        let currentState: string = states.start;
+        /** Error messages. */
+        const messages = {
+            noInit: `JSLL is not initialized with "awa.init(config)" function in <head>. Initialization script should be placed immediately after JSLL script.`,
+            notCallASAP: `"awa.init" is not called as soon as possible.`,
+            notImmediateAfter: `The JSLL init script should be immediately following the JSLL script.`,
+            notInHead: `The JSLL init script should be in <head>.`,
+            undefinedConfig: `The variable passed to "awa.init" is not defined.`
+        };
+
+        const inHead: Array<string> = [states.head, states.apiHead, states.headOtherScript];
+        const inBody: Array<string> = [states.body, states.apiBody, states. bodyOtherScript];
+        const otherScripts: Array<string> = [states.headOtherScript, states.bodyOtherScript];
```

Comment:
Can you add descriptions to this variables? Also the last element of `inBody` has an extra space that shouldn't be there `states.<space>bodyOtherScript`